src/components/common/text/TickerText.js

```diff
@@ -0,0 +1,119 @@
+import { EdgeText } from '../../themed/EdgeText'
+
+type Props = {
+  walletId: string,
+  style?: StyleSheet.styles,
+  tokenId: string
+}
+
+type GetDifferenceParams = {
+  exchangeRate: string,
+  currencyCode: string,
+  exchangeRates: GuiExchangeRates,
+  fiatExchangeRate: string
+}
+
+const getYesterdayDateRoundDownHour = () => {
+  const date = new Date()
+  date.setMinutes(0)
+  date.setSeconds(0)
+  date.setMilliseconds(0)
+  const yesterday = date.setDate(date.getDate() - 1)
+  return new Date(yesterday).toISOString()
+}
+
+const getPercentDeltaString = ({ currencyCode, exchangeRate, exchangeRates, fiatExchangeRate = '0' }: GetDifferenceParams) => {
+  // Create the default empty result function
+  const result = (value, deltaColorStyle = 'neutral') => ({
+    percentString: value != null ? `${value}%` : '',
+    deltaColorStyle
+  })
+
+  // Yesterdays Exchange Rate
+  const currencyPair = `${currencyCode}_iso:USD_${getYesterdayDateRoundDownHour()}`
+  const yesterdayUsdExchangeRate = toFixed(exchangeRates[currencyPair] ?? '0', 0, 2)
+  const yesterdayExchangeRate = mul(yesterdayUsdExchangeRate, fiatExchangeRate)
+
+  // Return the Exchange Rate without `percentageString` in case we are missing yesterday's rate
+  if (zeroString(yesterdayExchangeRate)) return result()
+  // Calculate the percentage difference in rate between yesterday and today
+  const differenceYesterday = sub(exchangeRate, yesterdayExchangeRate)
+  const differencePercentage = mul(div(differenceYesterday, yesterdayExchangeRate, 3), '100')
+
+  // Return zero result
+  if (zeroString(differencePercentage)) return result('0.00')
+
+  // If not zero, create the `percentageString`
+  const percentString = abs(differencePercentage)
+
+  // Return Positive result if greater then zero
+  if (gt(differencePercentage, '0')) return result(`+${percentString}`, 'positive')
+
+  // If it's not zero or positive, it must be a Negative result
+  return result(`-${percentString}`, 'negative')
+}
+
+/**
+ * Returns a text string that displays the crypto-fiat exchange rate and the
+ * daily % change from a wallet asset
+ **/
+export const TickerText = ({ walletId, style, tokenId }: Props) => {
+  const dispatch = useDispatch()
+  const styles = getStyles(useTheme())
+
+  const fiatCurrencyCode = useSelector(state => state.core.account.currencyWallets[walletId].fiatCurrencyCode)
+  const currencyInfo = useSelector(state => state.core.account.currencyWallets[walletId].currencyInfo)
+  const tokenOrNativeCode = tokenId ?? currencyInfo.currencyCode
+  const exchangeDenomination = dispatch(getExchangeDenominationFromState(currencyInfo.pluginId, tokenOrNativeCode))
+
+  const cryptoExchangeMultiplier = exchangeDenomination.multiplier
+  const isoFiatCurrencyCode = fixFiatCurrencyCode(fiatCurrencyCode)
+  const rateKey = `${tokenOrNativeCode}_${isoFiatCurrencyCode}`
+  const exchangeRates = useSelector(state => state.exchangeRates)
```

Comment:
It's a bad idea to select the entire 'exchangeRates' since that would trigger a render to ALL the TickerTexts every time ANY of them updates. 

Instead, you should be more precise in your selection.

You should do something like this - 

```js

const [fiatCurrencyCode, setFiatCurrencyCode] = useState(edgeWallet.fiatCurrencyCode)
  const isoFiatCurrencyCode = fixFiatCurrencyCode(fiatCurrencyCode)
  const rateKey = `${currencyCode}_${isoFiatCurrencyCode}`
  const yesterdayUsdRateKey = `${currencyCode}_iso:USD_${getYesterdayDateRoundDownHour()}`

  const rawExchangeRate = useSelector(state => state.exchangeRates[rateKey])
  const usdExchangeRate = useSelector(state => state.exchangeRates[`iso:USD_${isoFiatCurrencyCode}`])
  const yesterdayUsdExchangeRate = useSelector(state => state.exchangeRates[yesterdayUsdRateKey])
  
  ```
  
  
  
  Look at this file for reference: https://github.com/EdgeApp/edge-react-gui/blob/cfa44ebcc86e4316be7123357fdecdc8243cd571/src/components/themed/WalletListCurrencyRow.js